code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1323,10 +1264,12 @@
 		component.electronics_damage = 0
 		component.brute_damage = 0
 		component.installed = COMPONENT_INSTALLED
-	if(!cell)
-		cell = new(src)
-	cell.maxcharge = max(15000, cell.maxcharge)
-	cell.charge = cell.maxcharge
+		if(C == "power cell")
```

Comment:
checking the name of the component instead of the type of the wrapped item is more dangerous